drivers/at86rf2xx/aes_spi.c

```diff
@@ -0,0 +1,432 @@
+/* Copyright (C) 2020 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_at86rf2xx
+ * @{
+ *
+ * @file
+ * @brief       Implementation of at86rf2xx SPI security module (AES)
+ *
+ * @author      Fabian Hüßler <fabian.huessler@ovgu.de>
+ * @}
+ */
+
+#include "xtimer.h"
+#include "periph/spi.h"
+#include "at86rf2xx_aes.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define AES_DEBUG(...)  DEBUG("[at86rf2xx_aes_spi]: "__VA_ARGS__)
+
+#define AT86RF2XX_CMD_SRAM_READ             (0b00000000)
+#define AT86RF2XX_CMD_SRAM_WRITE            (0b01000000)
+
+static inline
+void at86rf2xx_spi_get_bus(const at86rf2xx_t *dev)
+{
+    spi_acquire(dev->params.spi, dev->params.cs_pin, SPI_MODE_0, dev->params.spi_clk);
+}
+
+static inline
+void at86rf2xx_spi_release_bus(const at86rf2xx_t *dev)
+{
+    spi_release(dev->params.spi);
+}
+
+static inline
+uint8_t _aes_status(at86rf2xx_t *dev)
+{
+    spi_transfer_byte(dev->params.spi, dev->params.cs_pin, true,
+                      AT86RF2XX_CMD_SRAM_READ);
+    spi_transfer_byte(dev->params.spi, dev->params.cs_pin, true,
+                      AT86RF2XX_REG__AES_STATUS);
+    return spi_transfer_byte(dev->params.spi, dev->params.cs_pin, false, 0);
+}
+
+static inline
+uint8_t _aes_check(at86rf2xx_t *dev)
+{
+    uint8_t status;
+    if ((status = _aes_status(dev)) & AT86RF2XX_AES_STATUS_MASK__AES_ER) {
+        AES_DEBUG("ERROR\n");
+        return status;
+    }
+    while (!(status & AT86RF2XX_AES_STATUS_MASK__AES_DONE)) {
+        AES_DEBUG("status: %02x\n", status);
+        status = _aes_status(dev);
+    }
+    return status;
+}
+
+static inline
+void _aes_open_read(at86rf2xx_t *dev, uint8_t addr)
+{
+    spi_transfer_byte(dev->params.spi, dev->params.cs_pin, true,
+                      AT86RF2XX_CMD_SRAM_READ);
+    spi_transfer_byte(dev->params.spi, dev->params.cs_pin, true,
+                      addr);
+}
+
+static inline
+void _aes_open_write(at86rf2xx_t *dev, uint8_t addr)
```

Comment:
ditto

Reply:
I cannot reuse them because, they acquire / release the bus in each call and `at86rf2xx_sram_write()` ignores what is read from the transceiver, which is important for fast SPI.

Reply:
hmmm I would say that these are "presentation layer" functions (not specific to AES). Maybe they can be added as generic functions? What do you think?

Reply:
Generally I agree, they could be moved to `at86rf2xx_internal` because they are not related to the security module. But I am not sure if I should do this because nothing else uses it, so far.

On the one side it touches the actual driver, and I think this should be done, if some day someone decides to modify the `at86rf2xx` SPI functions in a way that they do not acquire / release the SPI bus for every call. Right now I see no benefit of moving them.

On the other side it doesn´t harm because it is a small change.

I would keep them in place for now.